.github/workflows/build-irods-http-api-ubuntu.yml

```diff
@@ -0,0 +1,62 @@
+name: iRODS HTTP API Builder
+on: [pull_request]
+jobs:
+  build:
+    name: Build - ${{ matrix.os }}, iRODS ${{ matrix.irods-version }}
+    strategy:
+      fail-fast: false
+      matrix:
+        os: [ ubuntu-20.04, ubuntu-22.04, ubuntu-24.04 ]
```

Comment:
These entries are not the same as what's used in irods/irods.

Are these container names? irods/irods uses `ubuntu:20.04.`
Is there a difference between `ubuntu-20.04` and `ubuntu:20.04?`
Where is the `container` property? Is that deprecated now?

Reply:
> Are these container names?

The names come from: https://github.com/actions/runner-images

> Where is the `container` property? Is that deprecated now?

`container` is not deprecated.

Using irods/irods as an example:
https://github.com/irods/irods/blob/a6d646c37423f4e8fc25b21d4e15c2025eaed984/.github/workflows/build-irods-ubuntu.yml#L6-L10

Here, I am changing `runs-on`, the github action runner, to then build on the runner.

In irods/irods, the same runner is used, but `container` is set, so a container is created in the runner that then runs the code.

Not sure if this is what you meant, but just in case, the particular name under `matrix` does not matter. `os` could be labeled `images-to-build-and-test-on` for example.

> Is there a difference between ubuntu-20.04 and ubuntu:20.04?

The runner image does have more on it than the base Ubuntu image, to allow for the running of various actions (e.g. ccache action).

Reply:
Got it.

So why do something different than the irods/irods repo?
Is what you're doing the recommended way of doing things?

_I'm just curious._

Reply:
In a discuss:

We will use the container images instead of the runner images.

Reasoning is to be consistent with other projects.

Reply:
Using container images.